src/utils/functions/getDonationLateStatus.ts

```diff
@@ -0,0 +1,49 @@
+import { DateTime } from 'luxon';
+
+enum ContactLateStatusEnum {
+  OnTime,
+  LateLessThirty,
+  LateMoreThirty,
+  LateMoreSixty,
+}
+
+function selectLaterDate(lateAt: string, pledgeStartDate: string): string {
+  const lateDate = DateTime.fromISO(lateAt);
+  const pledgeDate = DateTime.fromISO(pledgeStartDate);
+  return lateDate > pledgeDate ? lateAt : pledgeStartDate;
```

Comment:
One nice property of ISO dates is that they can be lexicographically compared. So the parsing step can be skipped.

```suggestion
  return lateAt > pledgeStartDate ? lateAt : pledgeStartDate;
```